instrumentation/apache-pulsar/apache-pulsar-2.8/javaagent/src/test/groovy/io/opentelemetry/javaagent/instrumentation/pulsar/v28/PulsarClientTest.groovy

```diff
@@ -0,0 +1,506 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pulsar.v28
+
+import io.opentelemetry.instrumentation.test.AgentInstrumentationSpecification
+import io.opentelemetry.instrumentation.test.asserts.SpanAssert
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes
+import org.apache.pulsar.client.admin.PulsarAdmin
+import org.apache.pulsar.client.api.Consumer
+import org.apache.pulsar.client.api.Message
+import org.apache.pulsar.client.api.MessageListener
+import org.apache.pulsar.client.api.Producer
+import org.apache.pulsar.client.api.PulsarClient
+import org.apache.pulsar.client.api.Schema
+import org.apache.pulsar.client.api.SubscriptionInitialPosition
+import org.junit.Assert
+import org.testcontainers.containers.PulsarContainer
+import org.testcontainers.utility.DockerImageName
+import spock.lang.Shared
+
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+import static io.opentelemetry.api.trace.SpanKind.CONSUMER
+import static io.opentelemetry.api.trace.SpanKind.INTERNAL
+import static io.opentelemetry.api.trace.SpanKind.PRODUCER
+
+class PulsarClientTest extends AgentInstrumentationSpecification {
```

Comment:
we are trying to (slowly) convert all of the tests in this repository from Groovy to Java (#7195), would be very grateful if you could conver this, but it's not required for merging, thx!